terraform/modules/api-gateway/main.tf

```diff
@@ -0,0 +1,184 @@
+  "dev"  = aws_lambda_alias.dev.arn
+}, var.lambda_alias, aws_lambda_alias.live.arn)}/invocations"
+}
+
+resource "aws_api_gateway_deployment" "deployment" {
+  depends_on = [
+    aws_api_gateway_method.proxy,
+    aws_api_gateway_integration.proxy_integration
+  ]
+  rest_api_id = aws_api_gateway_rest_api.api_gateway.id
+
+  triggers = {
+    redeployment = timestamp()
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_api_gateway_stage" "stage" {
+  stage_name    = var.api_gateway_stage_name
+  rest_api_id   = aws_api_gateway_rest_api.api_gateway.id
+  deployment_id = aws_api_gateway_deployment.deployment.id
+
+  description = "Stage for ${var.api_gateway_stage_name}"
+  variables = {
+    lambda_alias = var.lambda_alias
+  }
+
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+resource "aws_api_gateway_account" "account" {
+  cloudwatch_role_arn = aws_iam_role.api_gateway_cloudwatch_role.arn
+
+  depends_on = [
+    aws_iam_role.api_gateway_cloudwatch_role,
+    aws_iam_role_policy.api_gateway_cloudwatch_policy
+  ]
+}
+
+resource "aws_iam_role" "api_gateway_cloudwatch_role" {
+  name = "${var.prefix}-api-gateway-cloudwatch-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          Service = "apigateway.amazonaws.com"
+        },
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "api_gateway_cloudwatch_policy" {
+  role = aws_iam_role.api_gateway_cloudwatch_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents",
+          "logs:DescribeLogGroups",
+          "logs:DescribeLogStreams",
+          "logs:GetLogEvents",
+          "logs:FilterLogEvents"
+        ],
+        Resource = [
+          "arn:aws:logs:${var.region}:${data.aws_caller_identity.current.account_id}:log-group:/aws/apigateway/*"
+        ]
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "apigateway:GET",
+          "apigateway:PUT",
+          "apigateway:POST",
+          "apigateway:DELETE",
+          "apigateway:PATCH"
+        ],
+        Resource = aws_api_gateway_rest_api.api_gateway.execution_arn
+      }
+    ]
+  })
+}
+
+resource "aws_lambda_permission" "allow_api_gateway" {
+  statement_id = "AllowAPIGatewayInvoke"
+  action       = "lambda:InvokeFunction"
+  function_name = lookup({
+    "live" = aws_lambda_alias.live.arn,
+    "dev"  = aws_lambda_alias.dev.arn
+  }, var.lambda_alias, aws_lambda_alias.live.arn)
+  principal  = "apigateway.amazonaws.com"
+  source_arn = "arn:aws:execute-api:${var.region}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.api_gateway.id}/*/*/*"
+}
+
+output "api_gateway_lambda_arn" {
```

Comment:
Nitpick: There's quite a few different resources in this `main.tf` file.
I'd probably think about:
- Moving things like this into the dedciated `outputs.tf` file
- Moving variables into the `vars.tf` like the prefix one below
- Moving the IAM resources into a new dedicated `iam.tf` file
- Moving the lambda resources into a new dedicated `lambda.tf`

That way main becomes leaner and holds the resources that we wanna know about straight away

Reply:
Yes I agree. I've moved things around.